Simplenote/src/main/java/com/automattic/simplenote/TagsActivity.java

```diff
@@ -82,48 +119,96 @@ public void onCreate(Bundle savedInstanceState) {
             getSupportActionBar().setDisplayHomeAsUpEnabled(true);
         }
 
-        Simplenote application = (Simplenote) getApplication();
-        mTagsBucket = application.getTagsBucket();
-        mNotesBucket = application.getNotesBucket();
-
         mTagsList = findViewById(R.id.list);
-        mTagsAdapter = new TagsAdapter();
-        mTagsList.setAdapter(mTagsAdapter);
+        mTagsList.setAdapter(tagItemAdapter);
         mTagsList.setLayoutManager(new LinearLayoutManager(TagsActivity.this));
         View emptyView = findViewById(R.id.empty);
         mEmptyViewImage = emptyView.findViewById(R.id.image);
         mEmptyViewText = emptyView.findViewById(R.id.text);
-        checkEmptyList();
+        checkEmptyList(false);
```

Comment:
_don't_ check empty list?

Reply:
Check empty list while is not searching. Do you think is confusing?

Reply:
> Do you think is confusing?

is it not confusing to you? `false` does not convey to me the part about `while is not searching`
I know it's common to use these flags, but I find _boolean blindness_ to be a common confusing style.
it's not like `checkEmptyList` itself is that clear, but is there a way we could remove the anxiety
introduced by an unexplained boolean?

Reply:
Yes, I think we can create a specific method for each behavior and let the caller decide which method to call. Allow me to introduce the change.

Reply:
it looks like we're not even "checking" anything here, more like updating a label.

doodling…

```java
// as separate methods
public void labelNoTags() {
    setEmptyListImage(…);
    setEmptyListMessage(…);
}

public void labelNoTagSearchResults() {
    if (DisplayUtils…) { … } else { … }

    setEmptyListMessage(…);
}

// as a named value with inherent meaning

public enum TagListContext {
    AllTags,
    SearchResults
}

public void updateEmptyListLabel(TagListContext context) {
    switch (context) {
        case AllTags:
            setEmptyListImage(…);
            setEmptyListMessage(…);
            break;

        case SearchResults:
             …
    }
}
```

of course if we use the `enum` we wouldn't have to switch (no pun intended) from an `if` to a `switch`. we could leave everything else the same. thoughts?

Reply:
I implemented them as separate methods. Fixed in e15ef97222b8dfc62bfe4f03b2dcf4b615ab63ec.